app/models/work_package/pdf_export/gantt.rb

```diff
@@ -0,0 +1,1146 @@
+    end
+
+    # translates the work package dates to column dates
+    # will be overwritten by subclasses
+    # @param [Array<Date>] _range
+    # @return [Array<Date>]
+    def build_column_dates_range(_range)
+      [] # to be overwritten
+    end
+
+    # defines which header row to be shown (years, quarters, months, days)
+    # will be overwritten by subclasses
+    def header_row_parts
+      [] # to be overwritten
+    end
+
+    # for a column, get all work packages that are active on that date
+    # will be overwritten by subclasses
+    # @param [Date] _date
+    # @param [Array<WorkPackage>] _work_packages
+    # @return [Array<WorkPackage>]
+    def work_packages_on_date(_date, _work_packages)
+      [] # to be overwritten
+    end
+
+    # milestones are centered in the cell only for day view
+    # will be overwritten by subclasses
+    def milestone_position_centered?
+      false # to be overwritten
+    end
+
+    # Calculates the start offset of the work package in date column
+    # will be overwritten by subclasses
+    # @param [WorkPackage] _work_package
+    # @param [Date] _date
+    # @return [Float]
+    def calc_start_offset(_work_package, _date)
+      0 # to be overwritten
+    end
+
+    # Calculates the end offset of the work package in date column
+    # will be overwritten by subclasses
+    # @param [WorkPackage] _work_package
+    # @param [Date] _date
+    # @return [Float]
+    def calc_end_offset(_work_package, _date)
+      0 # to be overwritten
+    end
+
+  end
+
+  class GanttBuilderMonths < GanttBuilder
+    def build_column_dates_range(range)
+      range
+        .map { |d| Date.new(d.year, d.month, -1) }
+        .uniq
+    end
+
+    def header_row_parts
+      %i[years quarters months]
+    end
+
+    def work_packages_on_date(date, work_packages)
+      work_packages.select { |work_package| wp_on_month?(work_package, date) }
+    end
+
+    def calc_start_offset(work_package, date)
+      test_date = Date.new(date.year, date.month, 1)
+      start_date = work_package.start_date || work_package.due_date
+      return 0 if start_date <= test_date
+
+      width_per_day = @column_width.to_f / date.end_of_month.day
+      day_in_month = start_date.day - 1
+      day_in_month * width_per_day
+    end
+
+    def calc_end_offset(work_package, date)
+      end_date = work_package.due_date || Time.zone.today
+      test_date = Date.new(date.year, date.month, -1)
+      return 0 if end_date >= test_date
+
+      width_per_day = @column_width.to_f / test_date.day
+      day_in_month = end_date.day
+      @column_width - (day_in_month * width_per_day)
+    end
+
+    def wp_on_month?(work_package, date)
+      start_date = work_package.start_date || work_package.due_date
+      end_date = work_package.due_date || Time.zone.today
+      Range.new(Date.new(start_date.year, start_date.month, 1), Date.new(end_date.year, end_date.month, -1))
+           .include?(date)
```

Comment:
Range can be created using literal `..`, also unless there is a reason not to, it is better to use quicker `cover?` method which compares value with range start and end instead of `include?` which for such case will build the list of values in range and check if the checked value is among them:
```suggestion
      (Date.new(start_date.year, start_date.month, 1)..Date.new(end_date.year, end_date.month, -1))
           .cover?(date)
```

Reply:
Thanks for the insight!